paddle/fluid/pybind/eager.cc

```diff
@@ -155,52 +187,175 @@ void InitEagerTensorWithEagerTensor(EagerTensorObject* self,
-    auto temp = egr::EagerTensor(dense_tensor);
-    self->eager_tensor.set_impl(
-        temp.copy_to(pten::TransToPtenBackend(place), true).impl());
-    VLOG(4) << "Different place, do TensorCopy";
+    if (flag_kwargs && kws_map["value"] != NULL) {
+      numpy_value = py::object(py::handle(kws_map["value"]), true);
+    }
+  }
+
+  if (kw_order_map["place"] <= args_num) {
+    place = CastPyArg2Place(PyTuple_GET_ITEM(args, 1), 1);
+  } else {
+    if (flag_kwargs && kws_map["place"] != NULL) {
+      place = CastPyArg2Place(kws_map["place"], 0);
+    }
+  }
+
+  if (kw_order_map["persistable"] <= args_num) {
+    persistable = CastPyArg2AttrBoolean(PyTuple_GET_ITEM(args, 2), 2);
+  } else {
+    if (flag_kwargs && kws_map["persistable"] != NULL) {
+      persistable = CastPyArg2AttrBoolean(kws_map["persistable"], 0);
+    }
+  }
+
+  if (kw_order_map["zero_copy"] <= args_num) {
+    zero_copy = CastPyArg2AttrBoolean(PyTuple_GET_ITEM(args, 3), 3);
+  } else {
+    if (flag_kwargs && kws_map["zero_copy"] != NULL) {
+      zero_copy = CastPyArg2AttrBoolean(kws_map["zero_copy"], 0);
+    }
+  }
+
+  if (kw_order_map["name"] <= args_num) {
+    PyObject* name_obj = PyTuple_GET_ITEM(args, 4);
+    if (name_obj == Py_None) {
+      act_name =
+          egr::Controller::Instance().GenerateUniqueName("generated_tensor");
+    } else {
+      act_name = CastPyArg2AttrString(name_obj, 4);
+    }
+  } else {
+    if (flag_kwargs) {
+      if (kws_map["name"] == NULL) {
+        act_name =
+            egr::Controller::Instance().GenerateUniqueName("generated_tensor");
+      } else {
+        act_name = CastPyArg2AttrString(kws_map["name"], 0);
+      }
+    } else {
+      act_name =
+          egr::Controller::Instance().GenerateUniqueName("generated_tensor");
+    }
+  }
+
+  if (kw_order_map["stop_gradient"] <= args_num) {
+    stop_gradient = CastPyArg2AttrBoolean(PyTuple_GET_ITEM(args, 5), 5);
+
+  } else {
+    if (flag_kwargs) {
+      if (kws_map["stop_gradient"] != NULL) {
+        stop_gradient = CastPyArg2AttrBoolean(kws_map["stop_gradient"], 0);
+      }
+    }
+  }
+  EmptyEagerTensorInitializer(py_tensor_ptr, act_name, place, persistable,
+                              stop_gradient);
+  InitEagerTensorWithNumpyValue(py_tensor_ptr, numpy_value, zero_copy);
+}
+
+// initialize EagerTensor by EagerTensor or framework::Tensor (mix args and
+// kwargs) automatically
+void AutoInitEagerTensorByTensor(
+    EagerTensorObject* py_tensor_ptr,
+    std::unordered_map<std::string, PyObject*> kws_map, PyObject* args,
+    bool flag_kwargs, Py_ssize_t args_num, bool init_by_egr_tensor = true) {
+  // key_word lists: [value, place, name]
+  std::unordered_map<std::string, Py_ssize_t> kw_order_map{
+      {"value", 1}, {"place", 2}, {"name", 3}};
+
+  paddle::platform::Place place =
+      egr::Controller::Instance().GetExpectedPlace();
+  std::string act_name = "";
+
+  if (kw_order_map["place"] <= args_num) {
```

Comment:
it seems these code is duplicated make them one

Reply:
After refactoring, there is no duplicated code.